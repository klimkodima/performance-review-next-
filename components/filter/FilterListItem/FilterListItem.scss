.filter-list-item {
  display: flex;
  justify-content: space-between;
  width: 100%;

  &__filter-name {
    color: #818181;
  }

  &__delete-button {
    padding: 0;
    width: 20px;
    height: 20px;
    transform: rotate(45deg);
  }
}
