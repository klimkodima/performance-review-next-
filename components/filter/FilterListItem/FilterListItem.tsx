import { ReactElement, memo } from 'react';

import Typography from '@mui/material/Typography';
import ListItem from '@mui/material/ListItem';
import IconButton from '@mui/material/IconButton';

import styles from './filterListItem.module.css';

type FilterListItemPropsType = {
  name: string;
};

const FilterListItem = memo(
  ({ name }: FilterListItemPropsType): ReactElement => (
    <ListItem>
      <div className='filter-list-item'>
        <Typography className='filter-list-item__filter-name'>
          {name}
        </Typography>
        <IconButton className='filter-list-item__delete-button'>+</IconButton>
      </div>
    </ListItem>
  )
);

export default FilterListItem;

FilterListItem.displayName = 'FilterListItem';
