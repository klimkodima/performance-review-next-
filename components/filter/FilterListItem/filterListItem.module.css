.filter-list-item {
  display: flex;
  justify-content: space-between;
  width: 100%;
}

.filter-list-item__filter-name {
  color: #818181;
}

.filter-list-item__delete-button {
  padding: 0;
  width: 20px;
  height: 20px;
  transform: rotate(45deg);
}