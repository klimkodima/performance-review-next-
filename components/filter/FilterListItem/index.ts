export { default as FilterListItem } from './FilterListItem';
