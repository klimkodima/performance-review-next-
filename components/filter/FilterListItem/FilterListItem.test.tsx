import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import FilterListItem from './FilterListItem';

describe('FilterListItem', () => {
  test('Matches snapshot', () => {
    const tree = renderer.create(<FilterListItem name='Alpha' />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correct name', () => {
    render(<FilterListItem name='Alpha' />);
    expect(screen.getByText(/Alpha/)).toBeInTheDocument();
  });
});
