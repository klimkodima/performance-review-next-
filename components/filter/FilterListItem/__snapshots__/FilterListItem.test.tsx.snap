// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterListItem Matches snapshot 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
  disabled={false}
>
  <div
    className="filter-list-item"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 filter-list-item__filter-name css-ahj2mt-MuiTypography-root"
    >
      Alpha
    </p>
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium filter-list-item__delete-button css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      +
    </button>
  </div>
</li>
`;
